### PR TITLE
Add task scheduler input

### PR DESCRIPTION
**Pull request by daniellmorris**
# Add task scheduler input

## PR Details

- Add task scheduler input to the rules engine
- Add an example for how to use the input

### CODE REVIEW

File: src/core/inputs/helpers/task-scheduler.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+ */
+// Using ts-ignore becaue mocha doesn't seem to like that we defined types separately
+// @ts-ignore 
+import later, { ScheduleData, Timer } from '@breejs/later';
+// import PQueue from 'p-queue';
+import { ILogger } from '../../../types';
+
+export interface ScheduleConfig {
+  defaultPerTaskConcurrency?: number; // Default per task concurrency. (Default to Infinity)
+  defaultMaxPerTaskQueueLength?: number; // Default to Infinity
```
Comment (EricWasTakenMJD) [inline src/core/inputs/helpers/task-scheduler.ts:12]: Hm... let's explain more what these properties are for.

File: examples/src/example-scheduler-input.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+// Bring in other Application Specific dependencies
+const RuleOutputProviderFile = require('./providers/rule_output_file');
+const ruleClosures = require('./providers/rule_closures');
+const ruleCorpus = require('./providers/rule_corpus_scheduler');
+const logger = require('./providers/custom_logger');
+
+const coreScheduledInput = new CoreInputScheduler(
+  {
+    defaultPerTaskConcurrency: 1,
+    defaultMaxPerTaskQueueLength: 100,
```
Comment (EricWasTakenMJD) [inline examples/src/example-scheduler-input.js:30]: Hmm... let's explain more about what these properties are for.

File: src/core/inputs/helpers/task-scheduler.ts [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  timezoneConfig?: 'local' | 'utc'; // Local or utc time (Defaults to utc)
+  tasks: TaskConfig[]; // List of tasks
+}
+export interface TaskConfig {
+  name: string;
+  maxTaskQueueLength?: number | null; // Queue length for task queue
+  concurrency?: number | null; // Allowed, concurrency for this task
```
Comment (EricWasTakenMJD) [inline src/core/inputs/helpers/task-scheduler.ts:19]: Hmm... let's explain more about what these properties are for.

**Review by EricWasTakenMJD — COMMENTED** [review]
other than the few clarification comments, these look good.

File: README.md [hunk 4]
```diff
@@ -71,6 +71,7 @@ The following Core Inputs are available starting with version 2 of Rule Harveste
   response, or even an empty response.
 - CoreInputUdp - (Available with version 2.4.0) Starts a UDP server on one or more ports then passes any requests into the rules engine. No response is
   expected
+- CoreInputScheduler - (Available with version 2.5.0) Starts a task scheduler that periodically runs the rules engine. It allows input to be provided to distinguis various scheduled tasks.
```
Comment (valtech-le) [inline README.md:74]: typo "distinguis"

**Pull request merged**